Toolkit/ArcGISToolkit/AR/ArcGISARView.swift

```diff
@@ -166,71 +165,73 @@ public class ArcGISARView: UIView {
-    public func arScreenToLocation(screenPoint: AGSPoint) -> AGSPoint {
+    public func arScreenToLocation(screenPoint: CGPoint) -> AGSPoint {
         fatalError("arScreen(toLocation:) has not been implemented")
     }
     
-    /// Resets the device tracking, using `originCamera` if it's not nil or the device's GPS location via the locationManager.
+    /// Resets the device tracking, using `originCamera` if it's not nil or the device's GPS location via the location data source.
     public func resetTracking() {
         initialLocation = nil
         startTracking()
     }
-
-    /// Resets the device tracking, using the device's GPS location via the locationManager.
+    
+    /// Sets the initial transformation used to offset the originCamera.
     ///
-    /// - Returns: Reset operation success or failure.
-    public func resetUsingLocationServices() -> Bool {
-        fatalError("resetUsingLocationServices() has not been implemented")
+    /// - Parameter initialTransformation: The initial transformation for originCamera offset.
+    /// - Returns: Whether setting the `initialTransformation` succeeded or failed.
+    public func setInitialTransformation(initialTransformation: AGSTransformationMatrix) -> Bool {
+        fatalError("setInitialTransformation(initialTransformation:) has not been implemented")
     }
     
-    /// Resets the device tracking using a spacial anchor.
+    /// Sets the initial transformation used to offset the originCamera.  The initial transformation is based on an AR point determined via existing plan hit detection from `screenPoint`.  If an AR point cannot be determined, this method will return `false`.
     ///
-    /// - Returns: Reset operation success or failure.
-    public func resetUsingSpatialAnchor() -> Bool {
-        fatalError("resetUsingSpatialAnchor() has not been implemented")
+    /// - Parameter screenPoint: The screen point to determine the `initialTransformation` from..
+    /// - Returns: Whether setting the `initialTransformation` succeeded or failed.
+    public func setInitialTransformation(screenPoint: CGPoint) -> Bool {
+        fatalError("setInitialTransformationPlacingOriginOnPlane(screenPoint:) has not been implemented")
     }
-    
+
     /// Starts device tracking.
-    public func startTracking() {
-        if !isSupported {
-            didStartOrFailWithError(ARError(.unsupportedConfiguration))
-            return
-        }
-        
-        if originCamera != nil {
-            // We have a starting camera, so no need to start the location manager, just finalizeStart().
-            finalizeStart()
+    public func startTracking(_ completion: ((_ error: Error?) -> Void)? = nil) {
+        // We have a location data source that needs to be started.
+        if let locationDataSource = self.locationDataSource {
+            locationDataSource.start { [weak self] (error) in
+                if let error = error {
```

Comment:
I think you could simplify this slightly by only calling `completion` in one place:

```swift
if error == nil {
    self?.finalizeStart()
}
completion?(error)
```

Reply:
Perfect.